gremlin-python/src/main/python/gremlin_python/driver/protocol.py

```diff
@@ -67,15 +77,22 @@ def connection_made(self, transport):
         super(GremlinServerHTTPProtocol, self).connection_made(transport)
 
     def write(self, request_message):
-        content_type = str(self._message_serializer.version, encoding='utf-8')
-
+        accept = str(self._response_serializer.version, encoding='utf-8')
         message = {
-            'headers': {'content-type': content_type,
-                        'accept': content_type},
-            'payload': self._message_serializer.serialize_message(request_message),
+            'headers': {'accept': accept},
+            'payload': self._request_serializer.serialize_message(request_message)
+                if self._request_serializer is not None else request_message,
             'auth': self._auth
         }
 
+        # The user may not want the payload to be serialized if they are using an interceptor.
+        if self._request_serializer is not None:
+            content_type = str(self._request_serializer.version, encoding='utf-8')
```

Comment:
What should the content-type be if there is no serialization?

Reply:
Should there be a check after calling the interceptors that content-type is set?

Reply:
I think that is reasonable. I opted not to do it because a lot of this depends on the underlying HTTP client. In gremlin-python, you could technically use whatever client you want since the protocol and transport are abstracted. There are some clients that would probably add that value in for you (I believe aiohttp has a default as well). This is why I noted in the upgrade documentation that this is an advanced feature. The user needs to know the capabilities of the underlying client and test it out themselves. This is the same reason we don't check for the type of the payload.